bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/handler/EpsonProjectorHandler.java

```diff
@@ -0,0 +1,425 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EpsonProjectorHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pauli Anttila, Yannick Schaus - Initial contribution
+ * @author Michael Lobstein - Improvements for OH3
+ */
+@NonNullByDefault
+public class EpsonProjectorHandler extends BaseThingHandler {
+    private static final int DEFAULT_POLLING_INTERVAL_SEC = 10;
+
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorHandler.class);
+    private final SerialPortManager serialPortManager;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+    private Optional<EpsonProjectorDevice> device = Optional.empty();
+
+    private boolean isPowerOn = false;
+    private int pollingInterval = DEFAULT_POLLING_INTERVAL_SEC;
+
+    public EpsonProjectorHandler(Thing thing, SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String channelId = channelUID.getId();
+        if (command instanceof RefreshType) {
+            Channel channel = this.thing.getChannel(channelUID);
+            if (channel != null) {
+                updateChannelState(channel);
+            }
+        } else {
+            EpsonProjectorCommandType epsonCommand = EpsonProjectorCommandType.getCommandType(channelId);
+            sendDataToDevice(epsonCommand, command);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        EpsonProjectorConfiguration config = getConfigAs(EpsonProjectorConfiguration.class);
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (THING_TYPE_PROJECTOR_SERIAL.equals(thingTypeUID)) {
+            device = Optional.of(new EpsonProjectorDevice(serialPortManager, config));
+        } else if (THING_TYPE_PROJECTOR_TCP.equals(thingTypeUID)) {
+            device = Optional.of(new EpsonProjectorDevice(config));
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+
+        pollingInterval = config.pollingInterval;
+        device.ifPresent(dev -> dev.setScheduler(scheduler));
+        device.ifPresent(dev -> {
+            try {
+                dev.connect();
+            } catch (EpsonProjectorException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        });
```

Comment:
Since you connect automatically as part of the pollingJob, I don't think you need to attempt connecting here. But you should at least set the initial status to unknown so that the handler is marked as initialized.

```suggestion
        updateStatus(ThingStatus.UNKNOWN);
```

Reply:
fixed